beeline/pom.xml

```diff
@@ -276,6 +276,11 @@
             <artifactId>maven-shade-plugin.log4j2-cachefile-transformer</artifactId>
             <version>2.1</version>
           </dependency>
+          <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>${log4j2.version}</version>
+          </dependency>
```

Comment:
Why do we need to add this? Did we use log4j transitively?

Reply:
it is bringing log4j-core 2.1 version which is also has lot of CVE's (https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-core/2.1)
<img width="1321" alt="Screenshot 2024-05-29 at 4 45 22 PM" src="https://github.com/apache/hive/assets/49618756/acef811a-b17f-4bb6-837d-38023262d9dc">

Reply:
Then the scope of this PR is wider than what is currently described in the ticket. Please adapt the description accordingly. 

It seems that the PR wants to avoid also using log4j 2 versions lower than X. In that case, I think we will need to revisit the enforcement rules to cover that case as well otherwise we cannot tell if it is handled properly in all modules.

Reply:
Hi @zabetak, you are correct, I want to exclude log4j 2 versions lower than X.

I tried modifying the enforcement rules to prevent log4j-core less than 2.17.1 (based on https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-core) like this
`<exclude>org.apache.logging.log4j:log4j-core:(,2.17.1)</exclude>`

But there are few problems:
	1. Druid-handler still requires log4j 2.8.2 (coming transitively from druid-server).
	2. Even with the enforcement rule, the log4j-core 2.1 coming in beeline (attached in previous comment snapshot) is not getting captured with this and still passing. It seems the rule are enforced on maven dependency tree output and as this beeline log4j-core is not present in dependency tree, it is getting enforced.
	
What's next:
	1. I can remove this beeline change and update the PR. Then I think the scope of the PR and work done in PR is aligned for log4j:log4j
	2. In description explicitly mention only this beeline pom change.

Let me know if you have any better way to resolve this.

Reply:
The `log4j-core:2.1` version is a dependency of https://github.com/edwgiz/maven-shaded-log4j-transformer plugin. There seems to be a 1-to-1 mapping between the plugin and the log4j-core version thus it doesn't make sense to declare an explicit dependency to log4j-core. If we want to avoid the usage of the old log4j version we should consider changing the version of the transformer. Nevertheless this is out of the scope of the current PR so let's track it under another JIRA if needed.

The `log4j-core:2.1` version is used by the shade plugin during the Hive build but it is not a Hive dependency thus its not used by the product. Since it is not used by the product it is not captured by the enforcer rule. I don't know of a good way to ban plugin dependencies and not sure if it is necessary or worth the effort investing in such tooling. If a plugin is deemed unsafe then we should either ban the plugin usage or bump its version.